src/CC2Edit/CC2Edit.cpp

```diff
@@ -2537,14 +2537,49 @@ void CC2EditMain::onTestChips2()
         return;
     }
 #ifndef Q_OS_WIN
-    QString winePath = settings.value(QStringLiteral("WineExe")).toString();
-    if (winePath.isEmpty() || !QFile::exists(winePath)) {
-        // Try standard paths
-        winePath = QStandardPaths::findExecutable(QStringLiteral("wine"));
-        if (winePath.isEmpty() || !QFile::exists(winePath)) {
-            QMessageBox::critical(this, tr("Could not find WINE"),
-                    tr("Could not find WINE executable.\n"
-                       "Please configure WINE in the Test Setup dialog."));
+    static auto versionRegex = new QRegExp(QStringLiteral("\\d{10} (?:proton|experimental)-6\\.\\d+-\\d+\n?"));
+    static auto slrSubdir = new QRegExp(QStringLiteral("[a-zA-Z0-9]+_platform_\\d+\\.\\d+\\.\\d+"));
+    QString steamRoot = settings.value(QStringLiteral("SteamRoot")).toString();
+    if (steamRoot.isEmpty() || !QFile::exists(steamRoot)) {
+        // The one which Steam looks for by default
+        steamRoot = QDir::homePath() + QStringLiteral("/.local/share/Steam");
+        if (!QFile::exists(steamRoot)) {
+            QMessageBox::critical(this, tr("Could not find Steam Root"),
+                    tr("Could not find Steam Root directory.\n"
+                       "Please set the Steam Root in the Test Setup dialog."));
+            return;
+        }
+    }
+    // Steam has it's own python version bundled for this, but it also requires injecting library files somehow, it's not worth it
+    QString pythonExe = QStandardPaths::findExecutable(QStringLiteral("python3"));
+    if (pythonExe.isEmpty() || !QFile::exists(pythonExe)) {
+        QMessageBox::critical(this, tr("Could not find Python 3"),
+                tr("Could not find Python 3 executable.\n"
+                   "Try installing Python 3."));
+        return;
+    }
+
+    QString protonExe = settings.value(QStringLiteral("ProtonExe")).toString();
+    if (protonExe.isEmpty() || !QFile::exists(protonExe)) {
+        // Find a proton version from the Steam apps
+        QDir steamApps(steamRoot + QStringLiteral("/steamapps/common"));
+        for(const QString &appPath: steamApps.entryList(QDir::Dirs)) {
+            QFile versionFile(steamApps.filePath(appPath + QStringLiteral("/version")));
+            if (!versionFile.exists() || !versionFile.open(QIODevice::ReadOnly)) continue;            
+            QByteArray versionBytes = versionFile.readAll();
+            versionFile.close();
+            QString version = QString::fromUtf8(versionBytes);
+            if (versionRegex->exactMatch(version)) {
+                protonExe = steamApps.filePath(appPath) + QStringLiteral("/proton");
+                if (!pythonExe.isEmpty()) break;
+            }
+        }

```

Comment:
It seems like, if we have multiple versions of Proton installed (which is certainly possible), the one we'll pick might be non-deterministic.  Since we're reading the version file anyway, perhaps we could sort them and pick the newest?  The user can still override it in the settings if they want a specific version.